packages/api/src/router/additionalMaterials/helpers.ts

```diff
@@ -0,0 +1,183 @@
+import { generateAdditionalMaterialModeration } from "@oakai/additional-materials";
+import {
+  type GenerateAdditionalMaterialInput,
+  additionalMaterialsConfigMap,
+} from "@oakai/additional-materials/src/documents/additionalMaterials/configSchema";
+import { generateAdditionalMaterialObject } from "@oakai/additional-materials/src/documents/additionalMaterials/generateAdditionalMaterialObject";
+import { generatePartialLessonPlanObject } from "@oakai/additional-materials/src/documents/partialLessonPlan/generateLessonPlan";
+import { type PartialLessonContextSchemaType } from "@oakai/additional-materials/src/documents/partialLessonPlan/schema";
+import { performLakeraThreatCheck } from "@oakai/additional-materials/src/threatDetection/lakeraThreatCheck";
+import { isToxic } from "@oakai/core/src/utils/ailaModeration/helpers";
+import type { ModerationResult } from "@oakai/core/src/utils/ailaModeration/moderationSchema";
+import type { PrismaClientWithAccelerate } from "@oakai/db";
+import { aiLogger } from "@oakai/logger";
+
+import type { LooseLessonPlan } from "../../../../aila/src/protocol/schema";
+
+const log = aiLogger("additional-materials");
+
+interface GenerateAdditionalMaterialParams {
+  prisma: PrismaClientWithAccelerate;
+  userId: string;
+  input: GenerateAdditionalMaterialInput & {
+    lessonId?: string | null;
+  };
+}
+
+export async function generateAdditionalMaterial({
+  prisma,
+  userId,
+  input,
+}: GenerateAdditionalMaterialParams) {
+  log.info("Generating additional material");
+
+  const result = await generateAdditionalMaterialObject({
+    provider: "openai",
+    parsedInput: {
+      documentType: input.documentType,
+      context: input.context,
+    },
+    action: input.action,
+  });
+
+  if (!result) {
+    throw new Error("Failed to generate additional material");
```

Comment:
Is the error already logged->Sentry by this point? Otherwise looks like we could be losing information